frontend/src/auth/authContext.tsx

```diff
@@ -0,0 +1,160 @@
+import { createContext, useContext, useState, ReactNode, PropsWithChildren } from 'react';
+
+// TODO: move somewhere more appropriate or read cookie differently
+function getCookie(name: string): string | null {
+	const nameLenPlus = (name.length + 1);
+	return document.cookie
+		.split(';')
+		.map(c => c.trim())
+		.filter(cookie => {
+			return cookie.substring(0, nameLenPlus) === `${name}=`;
+		})
+		.map(cookie => {
+			return decodeURIComponent(cookie.substring(nameLenPlus));
+		})[0] || null;
+}
+
+export type Role = 'Manager' | 'Cashier' | 'Clerk' | 'Supplier';
+
+type AuthContextType = {
+  username?: string;
+  roles: Role[]; // TODO: make a Set?
+  login: (username: string, password: string) => Promise<void>;
+  checkLoginStatus: () => Promise<string>; // TODO: something a bit more sophisticated than string
+  getUser: () => Promise<void>;
+  logout: () => Promise<void>;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+type AuthProviderProps = PropsWithChildren;
+
+export const AuthProvider = ({ children }: AuthProviderProps) => {
+  const [username, setUsername] = useState<string | undefined>(undefined);
+  const [roles, setRoles] = useState<Role[]>([]);
+
+  const login = async (username: string, password: string) => {
+    const sessionId = "1010110000";
+    const loginUrl = `http://localhost:8081/auth/local/login?session=${sessionId}`;
+    const requestHeaders = {
+      "Content-Type": "application/json",
+    };
+    const requestBody = {
+      user: username,
+      passwd: password,
+    };
+
+    try {
+      const response = await fetch(loginUrl, {
+          method: "POST",
+          headers: requestHeaders,
+          body: JSON.stringify(requestBody),
+          credentials: "include",
+      });
+
+      const { name: resUsername, attrs: { roles: resRoles } } = await response.json();
+
+      if (!resUsername || !resRoles) {
+        console.error("LoginFailed");
+        return;
+      }
+
+      setUsername(resUsername);
+      setRoles(resRoles);
+
+      console.log("Login success");
+    } catch {
+      console.error("Login failed");
+    }
```

Comment:
Maybe we should show the errors to the user. Something in the UI (maybe a toast message) could be implemented.
```suggestion
   } catch (err: any) {
    console.error("Login error:", err);
    return { success: false, message: err.message };
  }
```

Reply:
There is a toast.tsx. Pop-up is used in a new reservation page. Might need to move the function to not duplicate code.